std-bits/base/src/main/java/org/enso/base/parser/Separators.java

```diff
@@ -0,0 +1,130 @@
+package org.enso.base.parser;
+
+import static org.enso.base.parser.NumberWithSeparators.isDigit;
+
+import java.nio.CharBuffer;
+
+public record Separators(char first, char second, int count, int endIdx, int lastSeparator) {
+  /** Check if the character is a separator. */
+  static boolean isSeparator(char c) {
+    return c == '.' || c == ',' || c == ' ' || c == '\'';
+  }
+
+  /**
+   * Strip out the specified separators and replace with just full stop for decimal. If any
+   * character other than a digit, thousands or decimal separator is encountered then return null.
+   * If multiple decimal separators are encountered then return null.
+   */
+  static CharSequence strip(CharSequence value, int idx, int endIdx, char thousands, char decimal) {
```

Comment:
nitpick but if we have `endIdx` then maybe we could rename to `startIdx` as well?